src/core/components/ModelDropdownExpanded.tsx

```diff
@@ -0,0 +1,38 @@
+import styles from './NavBar.module.css';
+import { getAllModels } from '../config/models';
+import { ModelConfig } from '../types/models';
+import { useAppContext } from '../context/useAppContext';
+
+interface ModelDropdownProps {
+  setDropdownOpen: (dropdownOpen: boolean) => void;
+}
+
+const ModelDropdownExpanded = ({ setDropdownOpen }: ModelDropdownProps) => {
+  const { handleModelChange } = useAppContext();
+
+  const handleModelClick = (model: ModelConfig) => {
+    handleModelChange(model.id);
+    setDropdownOpen(false);
+  };
```

Comment:
@downing034 - one change I made here is closing the dropdown when the user selects a model.